stdlib/SparseArrays/src/sparsematrix.jl

```diff
@@ -373,12 +373,12 @@ SparseMatrixCSC(M::AbstractMatrix{Tv}) where {Tv} = SparseMatrixCSC{Tv,Int}(M)
 SparseMatrixCSC{Tv}(M::AbstractMatrix{Tv}) where {Tv} = SparseMatrixCSC{Tv,Int}(M)
 function SparseMatrixCSC{Tv,Ti}(M::AbstractMatrix) where {Tv,Ti}
     @assert !has_offset_axes(M)
-    I = findall(x -> x != 0, M)
-    eltypeTiI = Ti[i[1] for i in I]
-    eltypeTiJ = Ti[i[2] for i in I]
-    eltypeTvV = Tv[M[i] for i in I]
-    return sparse_IJ_sorted!(eltypeTiI, eltypeTiJ, eltypeTvV, size(M)...)
+    C = findall(v->v != 0, M)
```

Comment:
Do we want to use `iszero` here?

Reply:
Yeah, eventually `v != zero(Tv)` as per #30580. But better to keep things separated, right?

Reply:
As long as you are simplifying this function, it wouldn't hurt to use `findall(!iszero, M)` in this PR